prime_server/zmq_helpers.hpp

```diff
@@ -25,11 +25,10 @@ struct message_t {
   explicit message_t(
       void* data,
       size_t size,
-      void (*free_function)(void*, void*) = [](void* ptr, void* /*hint*/) {
-        delete[] static_cast<unsigned char*>(ptr);
+      void (*free_function)(void*, void*) = [](void* p, void*) {
+        delete[] static_cast<unsigned char*>(p);
       });
-  explicit message_t(size_t size = 0);
-  void reset(size_t size = 0);
+  explicit message_t(size_t size = 0, const void* data = nullptr);
```

Comment:
this is the only material change. we had this default constructor here just for testing but it turned out to be nice to have a long lived zmq message lying aroudn to reply with that actually had data in it.